packages/typespec-ts/test/unit/responsesGenerator.spec.ts

```diff
@@ -286,6 +286,76 @@ describe("Responses.ts", () => {
       `
       );
     });
+
+    it("mixed custom error response and core error response", async () => {
```

Comment:
Is this what happened in purview ? I doubt why not asking them to be consistent ? And Modular will never work for cases like this, because we don't split input model and output model.

Reply:
yes, because purview mixed rpc operation and azure core template operations which introduced duplicate definitions.

Reply:
For purview case they would change to consistent error response I think and here is comment https://github.com/Azure/azure-rest-api-specs/pull/26702#discussion_r1414911972.

But from our side I prefer to fix this to generate as what the customer defines otherwise this will fail to build.

This would not be an issue for Modular because the model marked as @error won't be generated. Any body shape is acceptable when calling createRestError.

This would be an issue if these models happen in normal 2xx response body. But I haven't met this case yet.